src/index.js

```diff
@@ -63,13 +64,28 @@ module.exports = ({ types: t }) => {
     if (!argumentDecoder) return;
 
     const { dependencyList, factory } = argumentDecoder(node.expression.arguments);
+    const isDependencyArray = dependencyList && t.isArrayExpression(dependencyList);
 
-    if (!t.isArrayExpression(dependencyList) && !factory) return;
+    if (!isDependencyArray && !factory) return;
 
     const isFunctionFactory = isFunctionExpression(factory);
     const dependencyInjections = [];
 
-    if (dependencyList) {
+    if (factory && dependencyList && (!isFunctionFactory || !isDependencyArray)) {
+      // define/require call with unknown factory type and/or non-array litteral dependencies.
+      path.replaceWithMultiple(
+        createFactoryInvocationWithUnknownArgTypes(
+          path,
+          opts,
+          dependencyList,
+          factory,
+          isDefineCall,
+          node.expression.arguments.length
+        )
+      );
+      return;
```

Comment:
This early return ultimately short-circuits the existing code and I believe there are some dead code paths further down that won't be hit any longer -- I'd prefer if we could use the existing code paths and add the additional edge cases in there, since the code is already well tested and been used in the published package for several years now, but I understand why it's easier from an implementation perspective to do it this way. Let's leave it for now, but please create an issue linking here and assign it to me, so that there is a backlog item for cleaning up the duplicate/dead code paths, as I don't want to increase the scope of this PR any further.

Reply:
I've tweaked the conditional for entering the new code so that all the existing use cases still use the old code.  All the previously existing unit tests still run unmodified and the new unit tests only test cases that weren't previously handled.

Reply:
Thanks, I think this is a smart call just for extra safety 👍